configuration/configuration.go

```diff
@@ -655,6 +655,11 @@ type Proxy struct {
 
 	// Password of the hub user
 	Password string `yaml:"password"`
+
+	// TTL is the expiry time of the content and will be cleaned up when it expires
+	// if not set, defaults to 7 * 24 hours
+	// If set to zero, will never expire cache
+	TTL *time.Duration `yaml:"ttl,omitempty"`
```

Comment:
why it has to be a pointer to a Duration?

Reply:
The pointer is used to determine if this is set.
If not set we fall back to the old behaviour that defaulted to 7 days

Reply:
Another approach here could be to consider `0` to be "not configured" (use default), and use a sentinel value (`-1` is fairly common) to disable. Golang can be a bit of a pain here to detect "unset" vs "default value", and using a special value for this can help in that respect as well. From a UX perspective it may also make it more explicit that it's _deliberately_ set to "never expire".


Speaking of UX; while I appreciate the use of `time.Duration`, I'm somewhat concerned about tha UX

> disclaimer:
>
> I don't know how we handle other `time.Duration` in the config; I looked
> at the field descriptions, which are very scarce, and most don't even document
> accepted formats, defaults, and min/max values (where applicable)

- I've seen this happen in the Docker API, where some fields also accepted a `time.Duration` (or similar), and users _assumed_ numeric values were in _seconds_ (e.g., set it to `3600`, but now ending up with a value that is a multitude shorter than they expected
- ^^ from that perspective it would be good if "no unit" would mean something sensible, OR to set a sensible minimum duration, and produce an error when setting values below that.
- ^^ related; it's technically correct to document all accepted units (`ns`, `us`, `ms`, `s`, `m`, or `h`), but (somewhat depending on what "normal" values are), we could slightly rephrase and omit "unlikely to be used" units (`ns`, `us`, `ms`)

Reply:
```
	fmt.Println(time.Duration(-1))
	// Output:
	// -1ns
```

The default output is a bit confusing

If we want to use -1 it is better to define a new type to handle negative values, which replaces and is compatible with Duration and supports units such as `Day`, `Month` etc.

Reply:
I agree. Duration is duration = a time period - we shouldn't be looking for wild solutions where there is no place for one. Negative values can lead to very silly situations (I've experienced misinterpretations of negative time/duration vals in past - none of them was funny to resolve in the middle of the night). My suggestion is to stick with regular duration values.

Reply:
Can we define a sensible minimum? Setting 3600 to expect "1 hour" and  ending up with 3600 NanoSeconds definitely can be a bit of a footgun